infrastructures/docker/environment/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/dockerfile/DockerfileEnvironment.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment.dockerfile;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.eclipse.che.api.core.model.workspace.Warning;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalEnvironment;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalMachineConfig;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalRecipe;
+
+/** @author Sergii Leshchenko */
+public class DockerfileEnvironment extends InternalEnvironment {
+  public static final String TYPE = "dockerfile";
+
+  private final String dockerfile;
```

Comment:
Is it the content of a Dockerfile? If so, please, specify it in comment or the name

Reply:
Renamed variable and the corresponding methods to use `dockerfileContent`